app/src/main/kotlin/at/bitfire/davdroid/push/PushRegistrationWorker.kt

```diff
@@ -123,23 +136,61 @@ class PushRegistrationWorker @AssistedInject constructor(
         }
     }
 
-    override suspend fun doWork(): Result {
-        logger.info("Running push registration worker")
-
+    private suspend fun registerSyncable() {
         val endpoint = preferenceRepository.unifiedPushEndpoint()
 
+        // register push subscription for syncable collections
         if (endpoint != null)
-            for (collection in collectionRepository.getSyncableAndPushCapable()) {
+            for (collection in collectionRepository.getPushCapableAndSyncable()) {
```

Comment:
Could exclude the already registered collections here and prevent them from being re-registered unecessarily?